utils.py

```diff
@@ -0,0 +1,26 @@
+import torch
+import torch.nn as nn
+
+def cross_entropy_mse(pred_win_prob, neural_net_prob, self_play_winner, search_prob, parameters, c = 1.00):
+	'''
+	This function takes the pred_win_prob v and the neural network move probabilities p, and combines two loss functions: the MSE
+	of the v with the true winner of the self-played game z, and the cross entropy loss of the move probabilities and the search probablities from the MCTS
+	'''
+	
+
+	raise NotImplementedError
+
+def train(train_data):
```

Comment:
I'd prefer to use something like Pytorch-Lightning for the training infrastructure, but it's something we can talk about. Also, the training loop will be more involved than is normally the case since we're generating the data as we go through self-play, see the link in my comment in #3

Reply:
I haven't used Pytorch-Lighting, interested in looking into it though as seeing how that would affect the code.

Reply:
Philip, could you go more into why you prefer pytorch lightning for this? I don't really see any advantage to using it.